app/assets/javascripts/discourse/app/components/search-menu/search-term.js

```diff
@@ -0,0 +1,93 @@
+import Component from "@glimmer/component";
+import discourseDebounce from "discourse-common/lib/debounce";
+import { action } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
+import { isiPad } from "discourse/lib/utilities";
+import { inject as service } from "@ember/service";
+import {
+  DEFAULT_TYPE_FILTER,
+  SEARCH_INPUT_ID,
+  focusSearchButton,
+} from "discourse/components/search-menu";
+
+const SECOND_ENTER_MAX_DELAY = 15000;
+
+export default class SearchTerm extends Component {
+  @service search;
+  @service appEvents;
+
+  @tracked lastEnterTimestamp = null;
+
+  // make constant available in template
+  get inputId() {
+    return SEARCH_INPUT_ID;
+  }
+
+  @action
+  updateSearchTerm(input) {
+    // utilze discourseDebounce as @debounce does not work for native class syntax
+    discourseDebounce(
```

Comment:
It should work properly since https://github.com/discourse/discourse/commit/e08a0b509dc92a95c289d12f2475189ee378f93f. Are you still seeing issues?